src/UI/Art/SpriteArtist/MapTileDetailsArtist.java

```diff
@@ -77,34 +80,29 @@ private static void generateOverlay(GameMap map, XYCoord coord)
     terrainAttrs.add(new AttributeArtist(SpriteLibrary.MapIcons.SHIELD.getIcon(), terrain.getDefLevel()));
     if( loc.durability < 99 ) terrainAttrs.add(new AttributeArtist(SpriteLibrary.MapIcons.HEART.getIcon(), loc.durability));
 
-    // Get the unit image.
-    ArrayList<AttributeArtist> unitAttrs = new ArrayList<AttributeArtist>();
-    BufferedImage unitImage = null;
+    // Collect any unit and its cargo
+    ArrayList<ArrayList<AttributeArtist>> unitAttrCache = new ArrayList<>();
+    ArrayList<BufferedImage> unitImageCache = new ArrayList<>();
     if( null != unit )
     {
-      UnitSpriteSet uss = SpriteLibrary.getMapUnitSpriteSet(unit);
-      unitImage = uss.getUnitImage();
-
-      unitAttrs.add(new AttributeArtist(SpriteLibrary.MapIcons.HEART.getIcon(), unit.getHP()));
-      if( unit.model.needsFuel() )
-        unitAttrs.add(new AttributeArtist(SpriteLibrary.MapIcons.FUEL.getIcon(), unit.fuel));
-      if( unit.ammo >= 0 ) 
-        unitAttrs.add(new AttributeArtist(SpriteLibrary.MapIcons.AMMO.getIcon(), unit.ammo));
-      if( unit.getCaptureProgress() > 0)
-        unitAttrs.add(new AttributeArtist(SpriteLibrary.getCaptureIcon(unit.CO.myColor),
-            map.getEnvironment(coord).terrainType.getCaptureThreshold()-unit.getCaptureProgress()));
+      addDetailInfoFor(unitImageCache, unitAttrCache,
+                       map, coord, unit);
     }
 
     ///////////////////////////////////////////////////////////////
     // Calculate the size of the panel.
-    int terrainPanelW = tileSize * 3;
-    int unitPanelW = (unitAttrs.isEmpty() ? 0 : (tileSize * 2));
-    int panelW = terrainPanelW + unitPanelW;
-    
-    // Each attribute with a 1-px buffer, plus space for the tile itself.
-    int numAttrs = Math.max(terrainAttrs.size(), unitAttrs.size());
     int bufferPx = 3;
-    int panelH = numAttrs*iconSize+numAttrs+(tileSize*2)+bufferPx;
+    //            left buf          attr icon               tile     right buffer
+    int columnW = bufferPx + iconSize+ATTR_TEXT_SPACING + tileSize + bufferPx*2;
+    int panelW = columnW * (unitImageCache.size() + 1);
+    if( unitImageCache.size() > 0 )
+      panelW += bufferPx*2; // The panel gets N-1 right buffers
+
+    int numAttrs = terrainAttrs.size();
+    for( ArrayList<AttributeArtist> attrList : unitAttrCache )
+      numAttrs = Math.max(numAttrs, attrList.size());
+    //         upper buffer   tile   lower buffer   each attribute with a 1-px buffer
+    int panelH = iconSize + tileSize + bufferPx  +  numAttrs*(iconSize+1);
```

Comment:
It seems like including `iconSize`and also `numAttrs*(iconSize+1)` should put us over by one, but it doesn't. Is there a "default" attribute that we draw but isn't counted in `numAttrs`?

Reply:
You're putting the +1 on the wrong thing.
It's `iconsize+1`, i.e. "the icon with 1 pixel of buffer"

Reply:
Oh, I misread. The extra `iconSize` is (as mentioned in the comment) the upper buffer of the panel (above the unit/terrain image).

Reply:
Oooh, okay. Right. The tile art just overlaps into that space, which makes it less apparent, and is also why we need it.